src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-logic.cpp

```diff
@@ -0,0 +1,1048 @@
+    {
+        return Status::UnsupportedCommand;
+    }
+
+    Status status           = Status::Success;
+    uint16_t connectionID   = commandData.connectionID;
+    auto & transportOptions = commandData.transportOptions;
+
+    // Contraints check on incoming transport Options
+    Status transportOptionsValidityStatus = ValidateIncomingTransportOptions(transportOptions);
+
+    VerifyOrDo(transportOptionsValidityStatus == Status::Success, {
+        ChipLogError(Zcl, "HandleAllocatePushTransport[ep=%d]: TransportOptions of command data is not Valid", mEndpointId);
+        handler.AddStatus(commandPath, transportOptionsValidityStatus);
+    });
+
+    FabricIndex fabricIndex = handler.GetAccessingFabricIndex();
+
+    TransportConfigurationStorage * transportConfiguration = FindStreamTransportConnectionWithinFabric(connectionID, fabricIndex);
+
+    if (transportConfiguration == nullptr)
+    {
+        ChipLogError(Zcl, "HandleModifyPushTransport[ep=%d]: ConnectionID Not Found.", mEndpointId);
+        handler.AddStatus(commandPath, Status::NotFound);
+        return std::nullopt;
+    }
+
+    if (mDelegate->GetTransportBusyStatus(connectionID) == PushAvStreamTransportStatusEnum::kBusy)
+    {
+        ChipLogError(Zcl, "HandleModifyPushTransport[ep=%d]: Connection is Busy", mEndpointId);
+        handler.AddStatus(commandPath, Status::Busy);
+        return std::nullopt;
+    }
+
+    std::shared_ptr<TransportOptionsStorage> transportOptionsPtr{ new (std::nothrow) TransportOptionsStorage(transportOptions) };
+
+    if (transportOptionsPtr == nullptr)
+    {
+        ChipLogError(Zcl, "HandleModifyPushTransport[ep=%d]: Memory Allocation failed for transportOptions", mEndpointId);
+        handler.AddStatus(commandPath, Status::ResourceExhausted);
+        return std::nullopt;
+    }
+    // Call the delegate
+    status = mDelegate->ModifyPushTransport(connectionID, *transportOptionsPtr);
+
+    if (status == Status::Success)
+    {
+        transportConfiguration->SetTransportOptionsPtr(transportOptionsPtr);
+    }
+
+    handler.AddStatus(commandPath, status);
+
+    return std::nullopt;
+}
+
+std::optional<DataModel::ActionReturnStatus>
+PushAvStreamTransportServerLogic::HandleSetTransportStatus(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                                                           const Commands::SetTransportStatus::DecodableType & commandData)
+{
+    if (IsNullDelegateWithLogging(commandPath.mEndpointId))
+    {
+        return Status::UnsupportedCommand;
+    }
+
+    Status status                              = Status::Success;
+    DataModel::Nullable<uint16_t> connectionID = commandData.connectionID;
+    auto & transportStatus                     = commandData.transportStatus;
+
+    VerifyOrDo(transportStatus != TransportStatusEnum::kUnknownEnumValue, {
+        ChipLogError(Zcl, "HandleAllocatePushTransport[ep=%d]: Motion Time Control (MaxDuration) Constraint Error", mEndpointId);
```

Comment:
This is not HandleAllocatePushTransport, nor is does this have anything to do with motion time control.